extensions/amp-story/1.0/animation-types.js

```diff
@@ -36,7 +41,7 @@ export let StoryAnimationDimsDef;
  * @typedef {{
  *   duration: number,
  *   easing: (string|undefined),
- *   keyframes: !KeyframesOrFilterFnDef,
+ *   keyframes: (!WebKeyframesCreateFnDef|!WebKeyframesDef),
```

Comment:
IMHO, `function():WebKeyFramesDef` is generally a good API extension for amp-animation/builder themselves.

Reply:
Agreed, but it currently fulfills the specific purpose of back-filling the first frame before the animations engine is ready.

I'll keep this task separate, since I'd first like to remove the requirement for function evaluation from `amp-story` presets. Unsure what the set of arguments for a generic Keyframes function would be outside of Stories, though.

Reply:
First frame is indeed tricky. I currently always require it. And it quite reasonably might have to block the display until those can be calculated. Amp-animation allows backfilling the first frame if not specified, but I'm not certain it immediately applies it (I'd think not). In the editors I'm at this point I'm considering requiring the first frame to be applied by the editor when creating the story. Though that has its own negatives sometimes. Let's discuss it offline at some point.